analyzers/tests/SonarAnalyzer.UnitTest/Extensions/VisualBasic/InterpolatedStringExpressionSyntaxExtensionsTests.cs

```diff
@@ -0,0 +1,97 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2022 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.CodeAnalysis.VisualBasic;
+using Microsoft.CodeAnalysis.VisualBasic.Syntax;
+using SonarAnalyzer.Extensions;
+
+namespace SonarAnalyzer.UnitTest.Extensions.VisualBasic
+{
+    [TestClass]
+    public class InterpolatedStringExpressionSyntaxExtensionsTests
+    {
+        private const string CodeSnipet = @"
+Public Class C
+        Public Sub M(ByVal notConstant As Integer, ByVal notConstantString As String)
+            {0}
+        End Sub
+
+        Private Function Foo() As String
+            Return ""x""
+        End Function
+End Class";
+
+        [DataTestMethod]
+        [DataRow(@"Dim methodCall = $""{Foo()}""")]
+        [DataRow(@"Dim nestedMethodCall = $""{$""{$""{Foo()}""}""}""")]
+        [DataRow(@"Const constant As Integer = 1
+                   Dim mixConstantNonConstant = $""{notConstant}{constant}""")]
+        [DataRow(@"Const constant As Integer = 1
+                   Dim mixConstantAndLiteral = $""TextValue {constant}""")]
+        [DataRow(@"Const constant As Integer = 1
+                   Dim mix = $""{constant}{$""{Foo()}""}{""{notConstant}""}""")]
+        public void TryGetGetInterpolatedTextValue_Returns_False(string code)
```

Comment:
I'd rename this to something like `tryGetGetInterpolatedTextValue_MixedStrings_ReturnsFalse`  so that it follows the pattern 
 "method_input/state_result".
 
 (same for all the tests in this file).